src/lib/support/ThreadOperationalDataset.cpp

```diff
@@ -38,11 +38,11 @@ namespace Thread {
  */
 class ThreadTLV final
 {
+public:
     static constexpr uint8_t kLengthEscape = 0xff; ///< This length value indicates the actual length is of two-bytes length, which
                                                    ///< is not allowed in Thread Operational Dataset TLVs.
-
-public:
-    static constexpr uint8_t kMaxLength = kLengthEscape - 1;
+    static constexpr uint8_t kHeaderSize = 2;      ///< Size of the TLV header (Type + Length).
+    static constexpr uint8_t kMaxLength  = kSizeOperationalDataset - kHeaderSize;
```

Comment:
I'm a little confused by these length/size changes.  Below, we allow SetLength to set a value greater than this `kMaxLength` constant, right?  Is that expected?

Reply:
I think the old constant was simply wrong, since an individual TLV element can't be longer than the whole dataset. SetLength is just enforcing "we don't support multi-byte lengths", I changed it from `!=` to `<` because the argument changed from uint_8 to size_t. I guess I could check for `<= maxLength` instead.

This ThreadTLV type is just an internal helper, so it's somewhat arbitrary what's validated here vs in the OperationalDataset logic.